### PR TITLE
heretic blade teleports are now invoked by using them in-hand

### PR DESCRIPTION
**Pull request by ATH1909**
## About The Pull Request

Shattering a heretic blade as a heretic or a heretic monster to teleport a random, mostly safe location is now done by activating the blade in-hand instead of by pressing an action button. This information will be given to a heretic or a heretic monster if they examine a heretic blade.

Teleporting using a heretic blade now plays a glass shattering noise, produces a slightly different message (that factors in whether or not the teleportation attempt was successful), and uses the TELEPORT_CHANNEL_MAGIC teleportation channel.

Muggles can now shatter heretic blades, but will not be teleported upon shattering a blade.

## Why It's Good For The Game

I definitely didn't die as a heretic because I thought that this power was sensibly bound to "z" (or another reasonable activation method) during the several times that I tried to invoke it. Haha, nope, I am just a gamer with a perfect memory who can play around unintuitive design decisions and think to check the top left corner of their screen in the middle of combat for a new ability that's only present while you have your blade outside of a container ahhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh

uh, action button bad, game telling you important information about your abilities without you needing to dive through the wiki (or the code) for it good.

## Changelog
:cl: ATHATH
qol: Shattering a heretic blade as a heretic or a heretic monster to teleport a random, mostly safe location is now done by activating the blade in-hand instead of by pressing an action button. This information will be given to a heretic or a heretic monster if they examine a heretic blade.
soundadd: Teleporting using a heretic blade now plays a glass shattering noise and produces a slightly different message (that factors in whether or not the teleportation attempt was successful).
fix: Teleporting using a heretic blade is now considered to be a magical teleportation method by the game.
/:cl:

### CODE REVIEW

File: code/modules/antagonists/eldritch_cult/eldritch_items.dm [hunk 1]
```diff
@@ -32,38 +32,9 @@
...
-	var/turf/safe_turf = find_safe_turf(zlevels = sword.z, extended_safety_checks = TRUE)
-	do_teleport(holder,safe_turf,forceMove = TRUE)
-	to_chat(holder,"<span class='warning'>You feel a gust of energy flow through your body... the Rusted Hills heard your call...</span>")
-	qdel(sword)
-
-
 /obj/item/melee/sickly_blade
 	name = "\improper Sickly Blade"
-	desc = "A sickly green crescent blade, decorated with an ornamental eye. You feel like you're being watched..."
+	desc = "A sickly green crescent blade, decorated with an ornamental eye. You feel like you're being watched... A heretic (or a servant of one) can shatter this blade to teleport to a random, mostly safe location by activating it in-hand."
```
Comment (Fikou) [inline code/modules/antagonists/eldritch_cult/eldritch_items.dm:37]: this sucks, add it as a notice message given to you if you are a heretic on examine

**Comment by SuperNovaa41** [comment]
You should cause a stun and knockdown to any non heretic that shatters the blade. Similar to cult gear.

File: code/modules/antagonists/eldritch_cult/eldritch_items.dm (hunk 1 above)
Comment (necromanceranne) [reply code/modules/antagonists/eldritch_cult/eldritch_items.dm:37]: It also saves having to update all the children of the parent blade.

**Comment by MrMelbert** [comment]
fun fact; you can shatter a heretic blade hands-free if you hold it in your suit storage slot (heretic armor) by pressing the action button (this even works while stunned and cuffed)

so this is technically a nerf, but it's a good nerf.

**Comment by necromanceranne** [comment]
I do have some mild concerns with it popping early on accident due to lag. Byond has a weird bug where it'll double input a click on an item, which gets translated into an activation in-hand. For something like this, I'm happy for it to be an action button, because I honestly think the attack_hand proc should be reserved for something else that might see more frequent use.

**Comment by moocowswag** [comment]
Would make for a pretty sizeable nerf to the heretic paths that are able to easily get the armors that can store their blade in their suit storage slot and activating it there.

**Comment by ATH1909** [comment]
Just finished testing this, and it does indeed work (with correct grammar and so on). I'm gonna work on adding a glass shattering noise and the description thing that Fikou mentioned now.

**Comment by ATH1909** [comment]
> I do have some mild concerns with it popping early on accident due to lag. Byond has a weird bug where it'll double input a click on an item, which gets translated into an activation in-hand. For something like this, I'm happy for it to be an action button, because I honestly think the attack_hand proc should be reserved for something else that might see more frequent use.

I couldn't replicate this by spam-clicking a single spot in my backpack to rapidly equip and stow away my heretic blade; do you have some video evidence of it?

**Comment by ATH1909** [comment]
alright, now to test the new changes

**Comment by ATH1909** [comment]
> fun fact; you can shatter a heretic blade hands-free if you hold it in your suit storage slot (heretic armor) by pressing the action button (this even works while stunned and cuffed)
> 
> so this is technically a nerf, but it's a good nerf.

are you sure that it works while you're stunned/cuffed? the action button has `check_flags = AB_CHECK_HANDS_BLOCKED|AB_CHECK_IMMOBILE`.

**Comment by ATH1909** [comment]
image

**Comment by ATH1909** [comment]
Should I change the shattering messages to be bolded?

**Comment by MrDoomBringer** [comment]
bold

**Comment by ATH1909** [comment]
> bold

https://youtu.be/JY15dOYoRq0?t=4

**Comment by necromanceranne** [comment]
> I couldn't replicate this by spam-clicking a single spot in my backpack to rapidly equip and stow away my heretic blade; do you have some video evidence of it?

I don't, and it may actually be a latency issue then. I know that this isn't something exclusive to just me and is a problem that people with even better connections to servers have experienced. I've even used it to snatch items out of peoples hands when we have tried to pick up the same item, as an example, and they definitely picked it up from their end as well.

All in all, probably not a real concern if it's some kind of outlier experience I'm having hoo haa.

**Comment by JoshAdamPowell** [comment]
woah I totally didn't know you could do this

**Comment by MrMelbert** [comment]
> 
> 
> > fun fact; you can shatter a heretic blade hands-free if you hold it in your suit storage slot (heretic armor) by pressing the action button (this even works while stunned and cuffed)
> > so this is technically a nerf, but it's a good nerf.
> 
> are you sure that it works while you're stunned/cuffed? the action button has `check_flags = AB_CHECK_HANDS_BLOCKED|AB_CHECK_IMMOBILE`.

image
Yeah, still works while stunned/cuffed.

**Comment by necromanceranne** [comment]
utterly scuffed

**Comment by ATH1909** [comment]
> > > fun fact; you can shatter a heretic blade hands-free if you hold it in your suit storage slot (heretic armor) by pressing the action button (this even works while stunned and cuffed)
> > > so this is technically a nerf, but it's a good nerf.
> > 
> > 
> > are you sure that it works while you're stunned/cuffed? the action button has `check_flags = AB_CHECK_HANDS_BLOCKED|AB_CHECK_IMMOBILE`.
> 
> image
> image
> image
> 
> Yeah, still works while stunned/cuffed.

amazing.

does it work if you're holding something in your active hand?

File: code/modules/antagonists/eldritch_cult/eldritch_items.dm (hunk 1 above)
Comment (ATH1909) [reply code/modules/antagonists/eldritch_cult/eldritch_items.dm]: done (like a week or two ago)

**Review by Mothblocks — CHANGES_REQUESTED** [review]
i forgot to send this change oops

File: code/modules/antagonists/eldritch_cult/eldritch_items.dm [hunk 2]
```diff
@@ -116,6 +86,11 @@
 		else
 			eldritch_knowledge_datum.on_ranged_attack_eldritch_blade(target,user,click_parameters)
 
+/obj/item/melee/sickly_blade/examine(mob/user)
+	. = ..()
+	if(IS_HERETIC(user) || IS_HERETIC_MONSTER(user))
+		. += {"<span class='notice'><B>A heretic (or a servant of one) can shatter this blade to teleport to a random, mostly safe location by activating it in-hand.</B></span>"}
```
[the remark this answers is not in the document]
Comment (ATH1909) [reply code/modules/antagonists/eldritch_cult/eldritch_items.dm:92]: I copied it from cult code, but okay

**Pull request merged**